src/Bicep.Core/Parsing/Parser.cs

```diff
@@ -839,6 +862,15 @@ current.Type switch
             }, RecoveryFlags.None, TokenType.NewLine);
         }
 
+        private SyntaxBase ConditionalObject()
+        {
+            var keyword = this.ExpectKeyword(LanguageConstants.IfKeyword);
+            var conditionExpression = this.WithRecovery(() => this.ParenthesizedExpression(true), RecoveryFlags.None, TokenType.LeftBrace, TokenType.NewLine);
+            var @object = this.WithRecovery(this.Object, GetSuppressionFlag(conditionExpression), TokenType.NewLine);
```

Comment:
Mind adding a line to the invalid resources file for something like: `... if ({'a': b}.a == true) {`? It's invalid syntax, just curious to see what sort of error message results from the recovery logic here.

Reply:
Added a test case. It produces a error message like `expect a ) character...` at the end, which is not very helpful. I also tried a invalid parenthesized expression in a resource declartion, and got a similar error message.
![image](https://user-images.githubusercontent.com/16367959/101568421-656de080-3987-11eb-97df-da0a2941f32a.png)